src/app/SimpleSubscriptionResumptionStorage.h

```diff
@@ -0,0 +1,149 @@
+/*
+ *    Copyright (c) 2023 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *      This file defines the CHIP CASE Session object that provides
+ *      APIs for constructing a secure session using a certificate from the device's
+ *      operational credentials.
+ */
+
+#pragma once
+
+#include <app/SubscriptionResumptionStorage.h>
+#include <lib/core/TLV.h>
+#include <lib/support/DefaultStorageKeyAllocator.h>
+
+namespace chip {
+namespace app {
+
+/**
+ * An example SubscriptionResumptionStorage using PersistentStorageDelegate as it backend.
+ */
+class SimpleSubscriptionResumptionStorage : public SubscriptionResumptionStorage
+{
+public:
+    CHIP_ERROR Init(PersistentStorageDelegate * storage)
+    {
+        VerifyOrReturnError(storage != nullptr, CHIP_ERROR_INVALID_ARGUMENT);
+        mStorage = storage;
+        return CHIP_NO_ERROR;
+    }
+
+    static StorageKeyName GetStorageKey(const ScopedNodeId & node);
```

Comment:
This should be an anonymous function in the `.cpp`. This would allow moving the dependency `lib/support/DefaultStorageKeyAllocator.h` to `SimpleSubscriptionResumptionStorage.cpp`. The public interface doesn't depend on this.